rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -172,137 +334,158 @@ def _create_features(
-            interpreter, RegexInterpreter
-        ):
-            interpreter = RegexInterpreter()
+        """Ignores the intepreter and returns features derived from custom encodings.
 
-        message = Message(data=sub_state)
-        # remove entities from possible attributes
-        attributes = set(
-            attribute for attribute in sub_state.keys() if attribute != ENTITIES
+        Compared to the regular SingleStateFeaturizer, the behaviour is the same as if
+        the interpreter fails to produce any features and `fallback_to_custom_encoding`
+        has been set to the named_attribute only.
+
+        Args:
+          substate:
+          interpreter: ignored
+          exclude: ignored
+          keep_only_sparse_seq_converted_to_sent: ignored
+          fallback_to_custom_encoding: ignored
+          sparse: Determines whether the generated feature is sparse.
+        """
+        name_attribute = get_name_attribute(sub_state)
+        if name_attribute:
+            return {
+                name_attribute: self._create_features(sub_state, name_attribute, sparse)
+            }
+
+        return {}
+
+
+class LabelTokenizerSingleStateFeaturizer(SingleStateFeaturizer):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__()
+        # it is hard to fully mimic old behavior, but SingleStateFeaturizer
+        # does the same thing if nlu pipeline is configured correctly
+        rasa.shared.utils.io.raise_deprecation_warning(
+            f"'{self.__class__.__name__}' is deprecated and "
+            f"will be removed in Rasa Open Source 3.0.0. "
+            f"It is recommended to use the '{SingleStateFeaturizer.__name__}' instead.",
+            docs=DOCS_URL_MIGRATION_GUIDE,
         )
 
-        parsed_message = interpreter.featurize_message(message)
-        output = self._get_features_from_parsed_message(parsed_message, attributes)
 
-        # check that name attributes have features
-        name_attribute = self._get_name_attribute(attributes)
-        if name_attribute and name_attribute not in output:
-            # nlu pipeline didn't create features for user or action
-            # this might happen, for example, when we have action_name in the state
-            # but it did not get featurized because only character level
-            # CountVectorsFeaturizer was included in the config.
-            output[name_attribute] = self._create_features(
-                sub_state, name_attribute, sparse
-            )
+##### TODO/FIXME:
+# - The following seems to be completely separate from the SingleStateFeaturizer
+# - it implements a very similar yet different featurization
 
-        return output
 
-    def encode_state(
-        self, state: State, interpreter: NaturalLanguageInterpreter
-    ) -> Dict[Text, List["Features"]]:
-        """Encode the given state with the help of the given interpreter.
+class EntityFeaturizer:
```

Comment:
Not sure I follow. Why do you want to implement a separate class here?

Reply:
It just helped me understand that the functionality that follows is completely independent of the rest (at least the functions itself). The logic seems to overlap and might be merged/solved with some of the above. No?